lib/breadcrumbs.ts

```diff
@@ -0,0 +1,182 @@
+import type { AsyncLocalStorage } from "async_hooks";
+import type { Breadcrumb, InternalBreadcrumb } from "./types";
+import type { Request, Response } from "express";
+const debug = require("debug")("raygun").bind(null, "[breadcrumbs]");
+
+let asyncLocalStorage: AsyncLocalStorage<InternalBreadcrumb[]> | null = null;
```

Comment:
I had a quick look into the man page for `AsyncLocalStorage` - but just to double-check: this is in-memory thread-local storage, correct? Like what I'd do with `ThreadLocal` in Java?

Reply:
Yes, not thread based, but rather the context is inside the `AsyncLocalStorage.run` function.
The `store` is indeed in memory, in this case an array of `InternalBreadcrumb`.